src/Chapter3.hs

```diff
@@ -971,6 +1115,26 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday deriving (Show, Eq, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend day = any ((== True) . (== day)) [Saturday,Sunday]
+
+nextDay :: WeekDay -> WeekDay
+nextDay Sunday = Monday
+nextDay d = succ d
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
  | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂